frontend/src/features/SidebarSuite/TabContent/UtilityOptionsSection/UtilityOptionsSection.tsx

```diff
@@ -33,73 +34,61 @@ import {
 } from "@mui/material";
 import { useAtomValue } from "jotai";
 
-const utilityComponents: UtilityOptions = {
-  download: {
-    callback: onDownload,
-    icon: FileDownloadIcon,
-  },
-  copyQueryTitle: {
-    callback: onCopyQueryTitle,
-    icon: LocalOfferOutlinedIcon,
-  },
-  copyQueryLink: {
-    callback: onCopyQueryLink,
-    icon: ShareOutlinedIcon,
-  },
-  emailQueryLink: {
-    callback: onEmailQueryLink,
-    icon: ForwardToInboxIcon,
-  },
-};
-
 export const UtilityOptionsSection = () => {
-  const { t } = useTranslation("settings");
+  const { t } = useTranslation(["settings", "common"]);
   const currentView = useAtomValue(currentDbViewAtom);
-  const {
-    fileName,
-    sourceLanguage,
-    queryParams,
-    pageSettings,
-    settingsOmissionsConfig,
-  } = useDbQueryParams();
+  const { fileName, dbLanguage } = useNullableDbRouterParams();
   let href: string;
 
-  if (typeof window !== "undefined") {
-    href = window.location.toString();
-  }
-
-  const { route } = useRouter();
+  const pageType = useResultPageType();
 
-  const isSearchRoute = route.startsWith("/search");
   const { download, error } = useDownloader();
 
   const [popperAnchorEl, setPopperAnchorEl] =
     useState<PopperAnchorState>(defaultAnchorEls);
 
+  const uiSettings = useMemo(() => {
+    if (pageType.isSearchPage) {
+      return utilityUISettings.filter(
+        (setting) => !UNAVAILABLE_SEARCH_PAGE_UI_UTILITIES.includes(setting),
+      );
+    }
+
+    if (dbLanguage && pageType.isDbFilePage) {
+      return getAvailableSettings<UtilityUISettingName>({
+        dbLanguage,
+        uiSettings: utilityUISettings,
+        unavailableSettingsForViewOrLang:
+          UNAVAILABLE_DB_SOURCE_PAGE_UI_UTILITIES[currentView],
+      });
+    }
+
+    // TODO: create error component
```

Comment:
is this for another PR? Maybe we can create an issue for this

Reply:
Done. #335